packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1353,7 +1371,9 @@ class PipelineOwner {
     assert(!_children.contains(child));
     assert(_debugAllowChildListModifications, 'Cannot modify child list after layout.');
     _children.add(child);
-    assert(_debugSetParent(child, this));
+    if (!kReleaseMode) {
```

Comment:
Do you really mean "not release", or do you mean "debug"?  This will run `_debugSetParent` in profile mode builds too.  There is also `kDebugMode` if that is really what you mean to say.

Reply:
I do mean to set the parent in profile mode as well because I wanted to specially mark timeline events from the root pipeline owner in the dev timeline (e.g. the LAYOUT, PAINT, etc. events in the timeline). To know whether a PipelineOwner is root, I am checking whether parent is null. And since the dev timeline is most useful in profile mode I opted to keep track of parents in profile mode as well.

Maybe it isn't necessary to give the root pipeline owner events special names in the dev timeline, though. I am not convinced it is anymore, actually. For example, for LAYOUT you will see one looooong LAYOUT bar in the timeline, with a sub-LAYOUT bars for each children stacked underneath them. It should be pretty clear what the root layout bar is if you roughly know how the pipeline owner tree works.

I originally added this because in the common "root pipeline owner with one child" case it looked a little strange in the timeline to have just two LAYOUT bars stacked on top of each other with approximately the same length. It looked like some event was duplicated on accident, so I changed the root bar to "LAYOUT (root)".

However, maybe that isn't necessary? Curious what people think.

Reply:
I like your timeline event suffix idea. If an app has a single view, it'll have two LAYOUT/PAINT/etc timeline events stacked on top of each other that are equally sized. A user might assume this is a bug if there isn't a suffix distinguishing these timeline events.

Did you consider using the view's ID as a suffix? For example imagine:

````
| --------------------- LAYOUT -----------------------|
| --- LAYOUT (view 1) --- | | --- LAYOUT (view 2) --- |
```

From my understanding, the view ID is available in all modes. If the root node isn't a `RenderView` (ie: `null` for root), you could omit the view ID suffix.

Reply:
> From my understanding, the view ID is available in all runtime modes. If the root node isn't a RenderView (ie: null for root), you could omit the view ID suffix.

I really like this idea. Sadly, the view ID isn't available to the PipelineOwner as it has no idea what the root node is. Even to check if its root node is a RenderView in debug mode only, we'd have to import view.dart into this file which I would consider a layer violation.

However: What if instead we attach the string representation of the root node? If it is a RenderView, it would include the viewID: `LAYOUT (RenderView#1234(view: FlutterView(id: 1))` - or would that be too verbose? We could also stuff that information into the argument to the event that the timeline would show in a separate window when you select the bar...

Reply:
> I do mean to set the parent in profile mode as well because I wanted to specially mark timeline events from the root pipeline owner in the dev timeline (e.g. the LAYOUT, PAINT, etc. events in the timeline). 

Okay, I just wanted to make sure you meant to run it in profile mode.  Seems like a legit reason to do so.

Reply:
> we'd have to import view.dart into this file which I would consider a layer violation.

Ah yup that's a good point.

> `LAYOUT (RenderView#1234(view: FlutterView(id: 1))`

That seems okay but I agree it's verbose. That said, I would still prefer this over having unlabeled children `LAYOUT`s.

A potential alternative could be that the `PipelineOwner` accepts a string description of its `rootNode` and use that for the suffixes. I don't feel strongly about this though, just food for thought.

Reply:
I think I am going to leave this as-is for now because I also realized that in profile mode RenderView.toString is not giving us that much useful information because we optimize out runtimeType.toString() calls for performance reasons. We could circumvent that, but I also don't want to increase the performance overhead of timeline tracing further...

We may have to revisit what to do here once we have a little more experience with using the timeline in multi view apps to see what information would actually be useful here...